refinery/analysis_manager/tasks.py

```diff
@@ -46,64 +48,240 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         analysis.set_status(Analysis.FAILURE_STATUS, error_msg)
 
 
-@task(base=AnalysisHandlerTask, max_retries=None)
-def run_analysis(analysis_uuid):
-    """Manage analysis execution"""
-    RETRY_INTERVAL = 5  # seconds
-
+def _get_analysis(analysis_uuid):
+    """
+    Try to fetch the Analysis from the given analysis_uuid. Fail the
+    `run_analysis` task if we cannot properly fetch it.
+    """
     try:
-        analysis = Analysis.objects.get(uuid=analysis_uuid)
-    except (Analysis.DoesNotExist, Analysis.MultipleObjectsReturned) as exc:
+        return Analysis.objects.get(uuid=analysis_uuid)
+    except (Analysis.DoesNotExist,
+            Analysis.MultipleObjectsReturned) as e:
         logger.error("Can not retrieve analysis with UUID '%s': '%s'",
-                     analysis_uuid, exc)
+                     analysis_uuid, e)
         run_analysis.update_state(state=celery.states.FAILURE)
         return
 
-    # if cancelled by user
-    if analysis.failed():
-        return
 
+def _get_analysis_status(analysis_uuid):
```

Comment:
Up to you but keep in mind that this breaks [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) which limits code reuse, makes code less readable, testable and maintainable. Difficulty of naming is just a symptom that simply points to this underlying problem.